### PR TITLE
[CURA-11537] Fix remove empty layers when layers do not equal layer height.

### PR DESCRIPTION
**Pull request by rburema**
This shouldn't happen, but neither should the hot-end ram into the build-plate (in fact, you want that even less).

_edit_ Unit tests are failing because of stuff outside of this ticket (prime-tower-enable), and github is not letting me update the branch for some reason.

### CODE REVIEW

File: src/FffPolygonGenerator.cpp [hunk 1]
```diff
@@ -769,12 +769,25 @@ bool FffPolygonGenerator::isEmptyLayer(SliceDataStorage& storage, const LayerInd
...
+    const auto get_highest_z = [&storage](const LayerIndex& layer_idx)
+    {
+        coord_t highest_z = 0;
+        for (std::shared_ptr<SliceMeshStorage>& mesh_ptr : storage.meshes)
+        {
+            auto& mesh = *mesh_ptr;
+            highest_z = layer_idx >= mesh.layers.size() ? highest_z : std::max(highest_z, mesh.layers[layer_idx].printZ);
+        }
+        return highest_z;
+    };
```
Comment (wawanbreton) [inline src/FffPolygonGenerator.cpp:781]: Could you avoid creating a lambda function for using it only once ?

File: src/FffPolygonGenerator.cpp [hunk 2]
```diff
@@ -769,12 +769,25 @@ bool FffPolygonGenerator::isEmptyLayer(SliceDataStorage& storage, const LayerInd
 
 void FffPolygonGenerator::removeEmptyFirstLayers(SliceDataStorage& storage, size_t& total_layers)
 {
+    const auto get_highest_z = [&storage](const LayerIndex& layer_idx)
+    {
+        coord_t highest_z = 0;
+        for (std::shared_ptr<SliceMeshStorage>& mesh_ptr : storage.meshes)
```
Comment (wawanbreton) [inline src/FffPolygonGenerator.cpp:775]: Use a const reference ?

**Review by wawanbreton — COMMENTED** [review]
Minor remarks (unit tests are fixed in a other branch)

File: src/FffPolygonGenerator.cpp (hunk 1 above)
Comment (rburema) [reply src/FffPolygonGenerator.cpp]: Yes, I can.

File: src/FffPolygonGenerator.cpp (hunk 2 above)
Comment (rburema) [reply src/FffPolygonGenerator.cpp]: Done!

**Pull request merged**